### PR TITLE
Implemented usage of existing address from db

### PR DESCRIPTION
**Pull request by gd-kkramarenko**
if posted address matches existing, existing address is used

### CODE REVIEW

File: EcommerceApp/backend/spring-boot-ecommerce-app/src/main/java/kkramarenko/ecommerceapp/service/CheckoutServiceImpl.java [hunk 1]
```diff
@@ -41,12 +47,36 @@ public PurchaseResponse placeOrder(Purchase purchase) {
...
+            Address savedBillingAddress = addressRepository.save(billingAddress);
+            order.setBillingAddress(savedBillingAddress);
+        }
+
+
         order.setStatus(OrderStatus.CREATED.toString());
 
 
         Customer customer = purchase.getCustomer();
+        System.out.println(customer);
```
Comment (gd-iistomin) [inline EcommerceApp/backend/spring-boot-ecommerce-app/src/main/java/kkramarenko/ecommerceapp/service/CheckoutServiceImpl.java:79]: Why this println is needed? Assume this is some debug logging that supposed to be removed.
For proper logging dedicated libraries such as log4j should be used

File: EcommerceApp/backend/spring-boot-ecommerce-app/src/main/java/kkramarenko/ecommerceapp/service/CheckoutServiceImpl.java [hunk 2]
```diff
@@ -41,12 +47,36 @@ public PurchaseResponse placeOrder(Purchase purchase) {
...
+
+        Address existingBillingAddress = addressRepository.findAddressByCityAndCountryAndStateAndStreetAndZipCode(
+                billingAddress.getCity(), billingAddress.getCountry(), billingAddress.getState(), billingAddress.getStreet(), billingAddress.getZipCode());
+
+        if (existingBillingAddress != null){
+            order.setBillingAddress(existingBillingAddress);
+        } else {
+            Address savedBillingAddress = addressRepository.save(billingAddress);
+            order.setBillingAddress(savedBillingAddress);
+        }
```
Comment (gd-iistomin) [inline EcommerceApp/backend/spring-boot-ecommerce-app/src/main/java/kkramarenko/ecommerceapp/service/CheckoutServiceImpl.java:72]: Can we make a method from that? These two code pieces are very similar

**Review by gd-iistomin — CHANGES_REQUESTED** [review]

File: EcommerceApp/backend/spring-boot-ecommerce-app/src/main/java/kkramarenko/ecommerceapp/service/CheckoutServiceImpl.java (hunk 1 above)
Comment (gd-kkramarenko) [reply EcommerceApp/backend/spring-boot-ecommerce-app/src/main/java/kkramarenko/ecommerceapp/service/CheckoutServiceImpl.java:79]: Removed

File: EcommerceApp/backend/spring-boot-ecommerce-app/src/main/java/kkramarenko/ecommerceapp/service/CheckoutServiceImpl.java (hunk 2 above)
Comment (gd-kkramarenko) [reply EcommerceApp/backend/spring-boot-ecommerce-app/src/main/java/kkramarenko/ecommerceapp/service/CheckoutServiceImpl.java:72]: Done

**Review by gd-iistomin — APPROVED** [review]

**Pull request merged**